R/PipeOpColRoles.R

```diff
@@ -57,52 +74,77 @@ PipeOpColRoles = R6Class("PipeOpColRoles",
   public = list(
     initialize = function(id = "colroles", param_vals = list()) {
       ps = ps(
-        # named list, each entry with a vector of roles
+        # named list, each entry with a vector of roles, names are columns
         new_role = p_uty(
           tags = c("train", "predict"),
           custom_check = crate(function(x) {
             first_check = check_list(x, types = "character", any.missing = FALSE, min.len = 1L, names = "named")
-            # return the error directly if this failed
-            if (is.character(first_check)) {
-              return(first_check)
-            }
-            # changing anything target related is not supported
-            # a value of "character()" will lead to the column being dropped
+            # Return the error directly if this failed
```

Comment:
see if giving the same col name twice with different rows actually assigns both roles, in which case 'names = "named"' is sufficient. otherwise set 'names = "unique"' here as well.